tests/integration/main.rs

```diff
@@ -0,0 +1,48 @@
+mod artichoke;
+
+use std::env;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+#[allow(dead_code)]
+#[derive(Debug)]
+struct CommandOutput<'a> {
+    call_args: Vec<&'a str>,
+    status: i32,
+    stdout: String,
+    stderr: String,
+}
+
+fn manifest_path() -> String {
+    env::var_os("CARGO_MANIFEST_DIR").unwrap().into_string().unwrap()
+}
+
+fn binary_path<'a>(name: &'a str) -> Result<PathBuf, String> {
+    let path = Path::new(&manifest_path()).join("target").join("debug").join(name);
+    match path.exists() {
+        true => Ok(path),
+        false => Err(format!("Can't find binary {} in ./target/debug/", name)),
+    }
+}
+
+fn run<'a>(binary_name: &'a str, call_args: Vec<&'a str>) -> Result<CommandOutput<'a>, String> {
+    let binary = binary_path(binary_name)?;
+
+    let output = Command::new(&binary)
```

Comment:
The deref here is not necessary since `Command::new` takes `AsRef<OsStr>` and `binary is not used elsewhere in this function.

```suggestion
    let output = Command::new(binary)
```